frontend/src/components/MapView/index.tsx

```diff
@@ -215,9 +215,21 @@ const MapView = memo(({ classes }: MapViewProps) => {
   const addMissingLayers = useCallback(() => {
     missingLayers.forEach(layerId => {
       const layer = LayerDefinitions[layerId as LayerKey];
+      const { serverLayerName } = layer as any;
```

Comment:
Why do we need the same logic in two places?

Reply:
ok, maybe create a util function?

Reply:
Done